python/packages/autogen-agentchat/src/autogen_agentchat/messages.py

```diff
@@ -28,89 +80,235 @@ class BaseMessage(BaseModel, ABC):
 
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
+    @abstractmethod
+    def content_to_text(self) -> str:
+        """Convert the content of the message to a string-only representation
+        that can be rendered in the console and inspected by the user.
+
+        This is not used for creating text-only content for models.
+        For :class:`ChatMessage` types, use :meth:`content_to_model_text` instead."""
+        ...
 
-class BaseChatMessage(BaseMessage, ABC):
-    """Base class for chat messages."""
+    @abstractmethod
+    def content_to_model_text(self) -> str:
+        """Convert the content of the message to text-only representation.
+        This is used for creating text-only content for models.
 
-    pass
+        This is not used for rendering the message in console. For that, use
+        :meth:`~BaseMessage.content_to_text`.
 
+        The difference between this and :meth:`content_to_model_message` is that this
+        is used to construct parts of the a message for the model client,
+        while :meth:`content_to_model_message` is used to create a complete message
+        for the model client.
+        """
+        ...
 
-class BaseAgentEvent(BaseMessage, ABC):
-    """Base class for agent events."""
+    @abstractmethod
+    def content_to_model_message(self) -> UserMessage:
+        """Convert the message content to a :class:`~autogen_core.models.UserMessage`
+        for use with model client, e.g., :class:`~autogen_core.models.ChatCompletionClient`."""
+        ...
 
-    pass
 
+class TextChatMessage(ChatMessage, ABC):
+    """Base class for all text-only :class:`ChatMessage` types.
+    It has implementations for :meth:`content_to_text`, :meth:`content_to_model_text`,
+    and :meth:`content_to_model_message` methods.
 
-class TextMessage(BaseChatMessage):
-    """A text message."""
+    Inherit from this class if your message content type is a string.
+    """
 
     content: str
     """The content of the message."""
 
-    type: Literal["TextMessage"] = "TextMessage"
+    def content_to_text(self) -> str:
+        return self.content
+
+    def content_to_model_text(self) -> str:
+        return self.content
+
+    def content_to_model_message(self) -> UserMessage:
+        return UserMessage(content=self.content, source=self.source)
+
+
+class AgentEvent(BaseMessage, ABC):
+    """Base class for agent events.
+
+    .. note::
+
+        If you want to create a new message type for signaling observable events
+        to user and application, inherit from this class.
+
+    Agent events are used to signal actions and thoughts produced by agents
+    and teams to user and applications. They are not used for agent-to-agent
+    communication and are not expected to be processed by other agents.
+
+    You should override the :meth:`content_to_text` method if you want to provide
+    a custom rendering of the content.
+    """
+
+    content: Any
```

Comment:
Is this necessary?

Reply:
if we want existing user code that depends on `ChatMessage` or `AgentEvent` to have a content field, then we need it.

Reply:
I moved `content` to concrete classes for now.